e2e/runtime_scan_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright © 2021 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"gotest.tools/assert"
+	"strings"
+	"testing"
+	"time"
+
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+
+	"github.com/openclarity/kubeclarity/api/client/models"
+	"github.com/openclarity/kubeclarity/e2e/common"
+)
+
+func TestRuntimeScan(t *testing.T) {
+	stopCh := make(chan struct{})
+	defer func() {
+		stopCh <- struct{}{}
+		time.Sleep(2 * time.Second)
+	}()
+	f1 := features.New("runtime scan").
+		WithLabel("type", "runtime").
+		Assess("runtime scan flow", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			t.Logf("Setup runtime env...")
+			assert.NilError(t, setupRuntimeScanTestEnv(stopCh))
+
+			t.Logf("start runtime scan...")
+			startRuntimeScan(t)
+
+			// wait for progress DONE
+			t.Logf("wait for scan done...")
+			assert.NilError(t, waitForScanDone(t))
+
+			t.Logf("get runtime scan results...")
+			results := common.GetRuntimeScanResults(t, kubeclarityAPI)
+			assert.Assert(t, results.Counters.Resources > 0)
+			assert.Assert(t, results.Counters.Vulnerabilities > 0)
+			assert.Assert(t, results.Counters.Packages > 0)
+			assert.Assert(t, results.Counters.Applications > 0)
+
+			assert.Assert(t, len(results.VulnerabilityPerSeverity) > 0)
+
+			assert.Assert(t, len(results.Failures) > 0)
```

Comment:
worth adding a comment here. e.g:
radial/busyboxplus:curl is not supported as it uses V1 image manifest, which should result in an error.